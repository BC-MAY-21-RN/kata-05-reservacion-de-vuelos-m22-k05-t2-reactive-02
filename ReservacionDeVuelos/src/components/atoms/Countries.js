import React from 'react';
import {View, Text, StyleSheet} from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';

const Countries = () => {
  return (
    <View style={styles.container}>
      <View style={styles.line}>
        <Text style={styles.abbreviation}>BEG</Text>
        <Icon name="airplane" style={styles.plane} />
        <Text style={styles.abbreviation}>AMS</Text>
      </View>
      <View style={styles.line}>
        <Text style={styles.country}>Serbia</Text>
        <Text style={styles.country}>Netherlands</Text>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    borderBottomWidth: 1,
    borderBottomColor: 'gray',
    marginHorizontal: 30,
    marginVertical: 10,
  },
  abbreviation: {
    fontSize: 28,
    color: 'black',
    fontWeight: 'bold',
  },
  country: {
    fontSize: 15,
    color: 'gray',
    paddingVertical: 3,
  },
  line: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  plane: {
    color: 'blue',
    fontSize: 25,
    marginTop: 15,
  },
  location: {
    fontSize: 10,
  },
});
export default Countries;
