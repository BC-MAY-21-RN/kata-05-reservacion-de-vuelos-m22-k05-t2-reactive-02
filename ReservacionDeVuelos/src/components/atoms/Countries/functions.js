const getText = (data, index, index2) => {
  if (index === 0 && index2 === 0) {
    return data.avOrigen;
  } else {
    if (index === 0 && index2 === 1) {
      return data.avDestino;
    } else {
      if (index === 1 && index2 === 0) {
        return data.origen;
      } else {
        return data.destino;
      }
    }
  }
};

const functions = {getText};

export default functions;
