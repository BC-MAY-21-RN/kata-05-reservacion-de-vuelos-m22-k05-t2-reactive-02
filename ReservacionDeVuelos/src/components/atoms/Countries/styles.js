import {StyleSheet} from 'react-native';

const styles = StyleSheet.create({
  container: {
    borderBottomWidth: 1,
    borderBottomColor: 'gray',
    marginHorizontal: 30,
    marginVertical: 10,
  },
  abbreviation: {
    fontSize: 28,
    color: 'black',
    fontWeight: 'bold',
  },
  country: {
    fontSize: 15,
    color: 'gray',
    paddingVertical: 3,
  },
  line: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  plane: {
    color: '#5C6DF8',
    fontSize: 25,
    marginTop: 15,
    position: 'absolute',
    alignSelf: 'center',
  },
  location: {
    fontSize: 10,
  },
});

export default styles;
