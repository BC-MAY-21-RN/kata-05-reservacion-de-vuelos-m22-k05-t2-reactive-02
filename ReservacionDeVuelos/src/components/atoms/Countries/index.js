import React from 'react';
import {View, Text} from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';
import styles from './styles';

const Countries = ({data}) => {
  return (
    <View style={styles.container}>
      <Icon name="airplane" style={styles.plane} />
      <View style={styles.line}>
        <Text style={styles.abbreviation}>{data.avOrigen}</Text>
        <Text style={styles.abbreviation}>{data.avDestino}</Text>
      </View>
      <View style={styles.line}>
        <Text style={styles.country}>{data.origen}</Text>
        <Text style={styles.country}>{data.destino}</Text>
      </View>
    </View>
  );
};

export default Countries;
