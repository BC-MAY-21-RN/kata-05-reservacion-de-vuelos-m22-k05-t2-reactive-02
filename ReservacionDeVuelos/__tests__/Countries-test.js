import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Countries from '../src/components/atoms/Countries';

describe('Tests Countries component', () => {
  it('Countries snapshot', () => {
    const renderCountries = renderer
      .create(
        <Countries
          values={{
            from: {title: 'AWS', subtitle: 'Amazon web services'},
            to: {title: 'ML', subtitle: 'Mercado Libre'},
            date: '2022-06-16',
            passengers: 5,
          }}
        />,
      )
      .toJSON();
    expect(renderCountries).toMatchSnapshot();
  });
});
