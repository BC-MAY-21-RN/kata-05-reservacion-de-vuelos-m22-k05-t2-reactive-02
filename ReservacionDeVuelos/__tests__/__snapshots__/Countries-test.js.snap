// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests Countries component Countries snapshot 1`] = `
<View
  style={
    Object {
      "borderBottomColor": "gray",
      "borderBottomWidth": 1,
      "marginHorizontal": 30,
      "marginVertical": 10,
    }
  }
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 12,
        },
        Object {
          "alignSelf": "center",
          "color": "#5C6DF8",
          "fontSize": 25,
          "marginTop": 15,
          "position": "absolute",
        },
        Object {
          "fontFamily": "Ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    ï„†
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 28,
          "fontWeight": "bold",
        }
      }
    />
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 28,
          "fontWeight": "bold",
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "gray",
          "fontSize": 15,
          "paddingVertical": 3,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "gray",
          "fontSize": 15,
          "paddingVertical": 3,
        }
      }
    />
  </View>
</View>
`;
